### PR TITLE
Fixes #17923 - Add user default taxonomy by name

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by ares** [comment]
@shlomizadok please note that tests are failing

**Comment by shlomizadok** [comment]
@ares - fixed

File: app/models/taxonomies/location.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Location < Taxonomy
   before_destroy EnsureNotUsedBy.new(:hosts)
 
   has_many :location_parameters, :class_name => 'LocationParameter', :foreign_key => :reference_id, :dependent => :destroy, :inverse_of => :location
-  has_many :default_users,       :class_name => 'User',              :foreign_key => :default_location_id
+  has_many :default_users,       :class_name => 'User',              :foreign_key => :default_location_id, :inverse_of => :location
```
Comment (shlomizadok) [inline app/models/taxonomies/location.rb]: Should be `default_location`, will change when tests are finished. Please don't merge yet.

**Comment by tstrachota** [comment]
For the sake of consistency, shouldn't we fix the issue in hammer? I'm afraid this would be the only api endpoint where we would use `*_name` as an input parameter. We use only id's everywhere else and hammer/other api consumer is supposed to deal with translating names to ids itself.

**Comment by shlomizadok** [comment]
It's a good point @tstrachota. I'd argue that other clients / scripts may also benefit from this change.
But if you / others feel it should be done on the hammer side - let me know.

**Comment by ares** [comment]
I'd prefer hammer fox first for the consistency. I'm not against also improving API and allow searching by name but that should probably be added to all resources at once.

**Comment by tstrachota** [comment]
I agree. There's certainly benefit in using names as identifiers when they're globally unique. But let's make it consistent first and if we decide to extend the api we should do it for all resources.

**Pull request closed without merging**